### PR TITLE
Configure step function arn rather than automatically finding it

### PR DESCRIPTION
**Pull request by davidfurey**
## Why are you doing this?

State machine ARNs are now stable even when changing the machine, so there is no need for this additional logic

I have added the ARNs to the config, but have not finished testing in CODE

### CODE REVIEW

File: cloud-formation/cfn.yaml [hunk 1]
```diff
@@ -127,6 +127,7 @@ Resources:
                 - states:ListStateMachines
                 - states:StartExecution
                 - states:GetExecutionHistory
+                - states:DescribeStateMachine
                 Resource: arn:aws:states:*:*:*
```
Comment (jacobwinch) [inline cloud-formation/cfn.yaml]: I know you haven't changed this here, but should we be more restrictive about which Step Functions this app can trigger?

File: app/monitoring/StateMachineMonitor.scala [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import SafeLogger._
+
+class StateMachineMonitor(client: RegularContributionsClient, actorSystem: ActorSystem) {
+
+  def start(): Unit = {
+    implicit val ec = actorSystem.dispatcher
+    actorSystem.scheduler.schedule(5.seconds, 60.seconds) {
+      client.healthy.onComplete {
+        case Success(true) => SafeLogger.debug("Regular contributions state machine is healthy")
+        case Success(false) => SafeLogger.error(scrub"Regular contributions state machine is not available")
```
Comment (jacobwinch) [inline app/monitoring/StateMachineMonitor.scala]: I'm slightly worried about someone re-phrasing this and breaking the monitoring you've implemented here. 

Some things we could do to mitigate against this:

1. Add a comment to note that we have a metric with a dependency on this precise String. 
2. Make it a val and imply the above (in the name).
3. Make it look a bit more machine-y 'regular-contributions-step-function-unavailable' to help people fill in the gaps.

File: cloud-formation/cfn.yaml [hunk 3]
```diff
@@ -282,3 +282,29 @@ Resources:
...
+      LogGroupName: !Sub support-frontend-${Stage}
+    FilterPattern: "Regular contributions state machine is not available"
+    MetricTransformations:
+     MetricNamespace: !Sub support-frontend-${Stage}
+      MetricName: state-machine-unavailable
+      MetricValue: 1
+
+  StateMachineUnavailableAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
```
Comment (jacobwinch) [inline cloud-formation/cfn.yaml]: I would suggest applying the CreateProdResources condition once you're happy with the testing of this.

**Comment by jacobwinch** [comment]
Monitoring looks good - I've made a few small suggestions 👍

File: cloud-formation/cfn.yaml (hunk 1 above)
Comment (davidfurey) [reply cloud-formation/cfn.yaml]: That would be nice, although that will then need to be stored in the app config and in the cloudformation

**Pull request merged**